pkgs/development/python-modules/scipy/default.nix

```diff
@@ -82,7 +90,17 @@ buildPythonPackage rec {
     blas = numpy.blas;
   };
 
-  setupPyBuildFlags = [ "--fcompiler='gnu95'" ];
+  pypaBuildFlags = [
+    # Current release pins pybind11's patch version and ours is newer.
+    # Consider restoring build-time dependency check later
+    "--skip-dependency-check"
+
+    # Skip sdist
+    "--wheel"
+
+    "-Csetup-args=-Dblas=cblas"
+    "-Csetup-args=-Dlapack=lapacke"
```

Comment:
This uses the `.pc` file generated by `build-support/alternatives/blas/`. I am not sure if that's what we want to use, and I'm entirely not sure when `build-support/alternatives/blas` _should_ be used.

For comparison:

```console
❯ pkg-config mkl-dynamic-ilp64-gomp --libs
-L/nix/store/ryqzxrbdabpjxbjw97w8x6nj4fg0qhcx-mkl-2023.0.0.25398/lib -Wl,--no-as-needed -lmkl_intel_ilp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl
❯ pkg-config cblas --libs
-L/nix/store/byi7kk3jd4nblhwbzavf82pc5p4s34b1-blas-3/lib -lcblas
```

Note that MKL does not seem to distribute a `mkl.pc` file, instead it ships:

```console
❯ pkg-config --list-all
mkl-dynamic-ilp64-gomp mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-ilp64-tbb  mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-lp64-iomp  mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-lp64-seq   mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-ilp64-iomp mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-ilp64-seq  mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-lp64-gomp  mkl - Intel(R) oneAPI Math Kernel Library
mkl-dynamic-lp64-tbb   mkl - Intel(R) oneAPI Math Kernel Library
```

Reply:
AFAIU, `blas.override { blasProvider = mkl; }` is somehow a way to expose `mkl_rt` the "single dynamic library", and `libcblas.so` in that derivation is a copy of `libblas.so` from `mkl`? What I do not understand if scipy is expected to link any of MKL statically, and whether the `blas` switching mechanism supports static linkage.

@matthewbauer I see you have both on `blas` switching, and on blas/lapack in `numpy`. Any hints?